doc/download.xml

```diff
@@ -0,0 +1,103 @@
+which must be a string.
+<P/>
+The result is a record which has at least the component <C>success</C>,
+with value <K>true</K> if the download was successful and <K>false</K>
+otherwise.
+In the former case, the component <C>result</C> is bound,
+whose value is a string that contains the contents of the downloaded file.
+In the latter case, the component <C>error</C> is bound,
+whose value is a string that describes the problem.
+<P/>
+The function calls the methods stored in the global list
+<C>Download_Methods</C> until one of them is successful.
+Currently there are methods based on the &GAP; functions
+<Ref Func="DownloadURL" BookName="curl"/> and
+<Ref Func="SingleHTTPRequest" BookName="io"/>,
+and methods based on the external programs <C>wget</C> and <C>curl</C>.
+<P/>
+An optional record <A>opt</A> can be given.
+The following components are supported.
+<P/>
+<List>
+<Mark><C>target</C></Mark>
+<Item>
+  If this component is bound then its value must be a string
+  that is a local filename,
+  and the function writes the downloaded contents to this file;
+  the returned record does not have a <C>result</C> component in this case.
+</Item>
+<Mark><C>verifyCert</C></Mark>
+<Item>
+  If this component is bound and has the value <K>false</K>
+  then those download methods that are based on <C>curl</C> or <C>wget</C>
+  will omit the check of the server's certificate.
+
+  The same effect is achieved for all <Ref Func="Download"/> calls
+  by setting the user preference <C>DownloadVerifyCertificate</C>
+  (see <Ref Subsect="subsec-DownloadVerifyCertificate"/>) to <K>false</K>
+  and omitting the <C>verifyCert</C> component from <A>opt</A>.
+</Item>
+</List>
+<P/>
+<Example>
+<![CDATA[
+gap> url:= "https://www.gap-system.org/Packages/utils.html";;
+gap> res:= Download( url );;
+gap> res.success;
+true
+gap> IsBound( res.result ) and IsString( res.result );
+true
+gap> res:= Download( Concatenation( url, "xxx" ) );;
+gap> res.success;
+false
+gap> IsBound( res.error ) and IsString( res.error );
+true
+]]>
+</Example>
+</Description>
+</ManSection>
+
+<Subsection Label="subsec-DownloadVerifyCertificate">
+<Heading>User preference <C>DownloadVerifyCertificate</C></Heading>
+<Index Key="DownloadVerifyCertificate"><C>DownloadVerifyCertificate</C></Index>
+
+The value <K>true</K> (the default) means that the server's certificate
+is checked in calls of <Ref Func="Download"/>,
+such that nothing gets downloaded if the certificate is invalid.
+
+If the value is <K>false</K> then those download methods that are based on
+<C>curl</C> or <C>wget</C> will omit the check of the server's certificate.
+
+One can set the value of the preference to <C>val</C> via
+<Ref Func="SetUserPreference" BookName="ref"/>, by calling
+<C>SetUserPreference( "utils", "DownloadVerifyCertificate", val )</C>,
+and access the current value via 
+<Ref Func="UserPreference" BookName="ref"/>, by calling
+<C>UserPreference( "utils", "DownloadVerifyCertificate" )</C>.
+
```

Comment:
I think we should say a bit about the intention behind this

```suggestion

We recommend to leave this preference at its default value <K>true</K>.
But sometimes it can be necessary to change it, e.g. to work around issues
with old operating systems which may not be able to correctly verify new
certificates. In general it is better to update such a system, but if that is
not an option, then disabling certificate checks may be a good last resort.
```

Reply:
> to leave 

Should be 

> leaving